packages/loader/container-loader/src/deltaManager.ts

```diff
@@ -1186,9 +1192,14 @@ export class DeltaManager
 
         // if we have connection, and message is local, then we better treat is as local!
         assert(!this.connection || this.connection.details.clientId !== message.clientId ||
-            this.lastSubmittedClientId === message.clientId, "Not accounting local messages correctly");
+            this.lastSubmittedClientIds.includes(message.clientId), "Not accounting local messages correctly");
 
-        if (this.lastSubmittedClientId !== undefined && this.lastSubmittedClientId === message.clientId) {
+        const foundIndex = this.lastSubmittedClientIds.indexOf(message.clientId);
+        if (foundIndex >= 0) {
+            // Report if we're getting messages from a clientId older than the most recent
+            if (foundIndex !== this.lastSubmittedClientIds.length - 1) {
+                this.logger.sendTelemetryEvent({ eventName: "matchedNonPreviousClientId" });
```

Comment:
i think we should at least log it as an error, i'm debating if it should be a container error. we may be able to prevent corruption if we can cancel all outbound ops at the moment of detection. the downside in the user could loose local state.

Reply:
Changed it to an error.  Personally I think it would be smoother if we monitored telemetry for hits and didn't turn this into a container error (because I think if this issue were more prevalent we would hear about it more often).  Then if we get any bad signals we can re-evaluate.